tests/test_build_html.py

```diff
@@ -1139,3 +1139,12 @@ def test_html_sourcelink_suffix(app):
     assert '<a href="_sources/images.txt"' in content_images
     assert (app.outdir / '_sources' / 'otherext.foo').exists()
     assert (app.outdir / '_sources' / 'images.txt').exists()
+
+
+@pytest.mark.sphinx('html', testroot='html_entity')
+def test_html_entity(app):
+    app.builder.build_all()
+    content = (app.outdir / 'index.html').text()
+
+    assert '&nbsp;' not in content
+    assert '&rarr;' not in content
```

Comment:
The most important thing here is there are no htmlentities without `&lt;`, `&gt;`, `&amp;`, '&quot;` and `&apos;`.
So it is better to use regexp to ensure non-existence of htmlentities.